### ISSUE TITLE
Fix 'test_feeds/test_missing_fields' tests in 4.2

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#1553]
### Issue information

| Related issue |
|--|
| Closes https://github.com/wazuh/wazuh-qa/issues/1531|

When it comes to starting with the test fix, there is some information that can help you:
```
Module: Vulnerability Detector
Support Team:  "binary beasts"
Target: Manager   
OS: Linux
```
I add some information obtained while researched the logs that failed.

## Case 1:

|  Type | Description  |
|--|--|
|Test Path | test_feeds/msu/test_missing_fields_msu_feed
| Consistent | yes
| Test Execution | 3/3 executions failed
| Case | test_missing_fields_msu_feed
| Cases Fails | 7
| Summary | On `check_feed_imported_successfully` method is called. The problem seems to be a Parser error when reading the file.
| Error message |  "Could not find the message: 'Microsoft Security Update' feed finished successfully"

## Case 2:

|  Type | Description  |
|--|--|
|Test Path | test_feeds/msu/test_missing_fields_msu_feed
| Consistent | no
| Test Execution | 1/3 executions failed
| Case | test_no_feed_changes
| Cases Fails | 1
| Summary | When `check_feed_imported_successfully` is called. The problem seems to be a Parser error and it will be because there is a problem when imported a file.
| Error message |  "Could not find the message: 'Microsoft Security Update' feed finished successfully"

## Case 3:

|  Type | Description  |
|--|--|
|Test Path | test_feeds/redhat/test_missing_fields_redhat_feed
| Consistent | no
| Test Execution | 2/3 executions failed
| Cases Fails | 27
| Summary | It should to do research with more details because in the logs doesn't show any details.

In order to finish this issue the following tasks should be fulfilled:
- [ ] Research of fails.
- [ ] Apply Fix 
- [ ] Full Green/ Full Yellow  in `test_feeds/redhat/test_missing_fields_redhat_feed` for 3 times.
- [ ] Full Green/ Full Yellow  in `test_feeds/msu/test_missing_fields_msu_feed` for 3 times.
- [ ] Documentation of any important change done for these tests or the used tools.

**Comment by mdengra** [comment]
### 2021-07-13
After multiple tests, it appears that these errors originate in the `monitoring.py` module of the QA framework, specifically in the `QueueMonitor` class, which is the one that finally checks if the related logs have been written. 

These logs are written to the log file correctly, but the `QueueMonitor` class is not able to detect them when a large number of messages are written to the log file (**log flooding**). To verify this I have disabled the `sca` module, which generates a large number of logs and the tests involved have been running correctly.

I continue to perform checks to locate the specific source of the problem.

**Comment by damarisg** [comment]
This issue requires more research because when we execute it, we keep getting fails.  We can see that this test is inconsistency.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[Results1_MissingMSU.log](https://github.com/wazuh/wazuh-qa/files/6811706/Results1_MissingMSU.log) | 2021-07-13| Seyla | 🟢
| [Results2_MissingMSU.log](https://github.com/wazuh/wazuh-qa/files/6811707/Results2_MissingMSU.log)| 2021-07-13| Seyla | 🟢
| [Results3_MissingMSU.log](https://github.com/wazuh/wazuh-qa/files/6811715/Results3_MissingMSU.log)| 2021-07-13| Seyla | 🟢
| [Results1_MissingRedhat.log](https://github.com/wazuh/wazuh-qa/files/6811719/Results1_MissingRedhat.log) | 2021-07-13| Seyla | 🔴 
| [Results2_MissingRedhat.log](https://github.com/wazuh/wazuh-qa/files/6811721/Results2_MissingRedhat.log)| 2021-07-13| Seyla | 🔴 
| [Results3.1_MissingRedhat.log](https://github.com/wazuh/wazuh-qa/files/6811786/Results3.1_MissingRedhat.log)| 2021-07-13| Seyla | 🔴

**Comment by pereyra-m** [comment]
Results for `test_feeds/msu/test_missing_fields_msu_feed.py`

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[centos_test_missing_fields_msu_feed_1.log](https://github.com/wazuh/wazuh-qa/files/6812643/centos_test_missing_fields_msu_feed_1.log)| 2021-07-13| Matias | :red_circle:
| [centos_test_missing_fields_msu_feed_2.log](https://github.com/wazuh/wazuh-qa/files/6812642/centos_test_missing_fields_msu_feed_2.log)| 2021-07-13| Matias | :red_circle:
| [centos_test_missing_fields_msu_feed_3.log](https://github.com/wazuh/wazuh-qa/files/6812641/centos_test_missing_fields_msu_feed_3.log)| 2021-07-13| Matias | :red_circle:

**Comment by mdengra** [comment]
### 2021-07-14
I ran the tests again by recreating a new Vagrant box with a clean environment.

Test results with the default modules enabled in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_missing_fields_msu_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6814912/test_missing_fields_msu_feed_local_r1.log)| 2021-07-14| Miguel | :green_circle: 
|[test_missing_fields_msu_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6814917/test_missing_fields_msu_feed_local_r2.log)| 2021-07-14| Miguel | :green_circle: 
|[test_missing_fields_msu_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6814918/test_missing_fields_msu_feed_local_r3.log)| 2021-07-14| Miguel | :green_circle: 
|[test_missing_fields_redhat_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6814921/test_missing_fields_redhat_feed_local_r1.log)| 2021-07-14| Miguel | :red_circle: 
|[test_missing_fields_redhat_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6814922/test_missing_fields_redhat_feed_local_r2.log)| 2021-07-14| Miguel | :green_circle: 
|[test_missing_fields_redhat_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6814932/test_missing_fields_redhat_feed_local_r3.log)| 2021-07-14| Miguel | :green_circle: 


Test results after disabling the `sca` module:

**Comment by pereyra-m** [comment]
Results for `test_feeds/msu/test_missing_fields_msu_feed.py`
Now, the following modules were turned off: `sca`, `syscollector` and `rootcheck`

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_missing_fields_msu_feed_modules_off_1.log](https://github.com/wazuh/wazuh-qa/files/6816243/test_missing_fields_msu_feed_modules_off_1.log)| 2021-07-14| Matias | :green_circle:
| [test_missing_fields_msu_feed_modules_off_2.log](https://github.com/wazuh/wazuh-qa/files/6816242/test_missing_fields_msu_feed_modules_off_2.log)| 2021-07-14| Matias | :green_circle:
| [test_missing_fields_msu_feed_modules_off_3.log](https://github.com/wazuh/wazuh-qa/files/6816241/test_missing_fields_msu_feed_modules_off_3.log)| 2021-07-14| Matias | :green_circle:

**Comment by damarisg** [comment]
Results when disabling modules: `sca`, `syscollector` and `rootcheck`.


| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[MissingFieldMsu1.log](https://github.com/wazuh/wazuh-qa/files/6818257/MissingField1.log)| 2021-07-14| Seyla| 🟡 
|[MissingField2.log](https://github.com/wazuh/wazuh-qa/files/6818255/MissingField2.log)| 2021-07-14| Seyla| :green_circle:
| [MissingField3.log](https://github.com/wazuh/wazuh-qa/files/6818256/MissingField3.log)| 2021-07-14| Seyla| :green_circle:
| [MissingFieldRedHat2.log](https://github.com/wazuh/wazuh-qa/files/6818252/MissingFieldRedHat2.log) | 2021-07-14| Seyla| 🟡 
|[MissingFieldRedHat3.log](https://github.com/wazuh/wazuh-qa/files/6818253/MissingFieldRedHat3.log)| 2021-07-14| Seyla| :green_circle:
|[MissingFieldRedHat1.log](https://github.com/wazuh/wazuh-qa/files/6818254/MissingFieldRedHat1.log)| 2021-07-14| Seyla| :green_circle:


| Reference | Status | 
|--|--|
| 🟢 | Pass without warnings|
| 🟡 | Pass with Warnings |

**Comment by damarisg** [comment]
This issue is closed, after executes for a lot of times and see that works. The solved applied was disabling `sca`, `syscollector` and `rootcheck` modules.

**Comment by mdengra** [comment]
### 2021-07-22

Used Wazuh-QA branch: `1531-full-yellow-vuln-det`
Test results with the default settings in the `ossec.conf`: 

### Case 1, 2
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_missing_fields_msu_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6863125/test_missing_fields_msu_feed_local_r1.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_missing_fields_msu_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6863126/test_missing_fields_msu_feed_local_r2.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_missing_fields_msu_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6863136/test_missing_fields_msu_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle: 


### Case 3
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_missing_fields_redhat_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6863144/test_missing_fields_redhat_feed_local_r1.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_missing_fields_redhat_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6863147/test_missing_fields_redhat_feed_local_r2.log)| 2021-07-22 | Miguel | :red_circle: 
|[test_missing_fields_redhat_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6863148/test_missing_fields_redhat_feed_local_r3.log)| 2021-07-22 | Miguel | :red_circle: 

Note that the same use case (first case) fails in the two first runs:
```
FAILED test_vulnerability_detector/test_feeds/redhat/test_missing_fields_redhat_feed.py::test_invalid_redhat_feed[REDHAT_configuration-missing: generator]
```

**Comment by mdengra** [comment]
#### Same scenario as in the issue: https://github.com/wazuh/wazuh-qa/issues/1548
The problem is related to the issue: #1602, in which FileMonitor does not detect a message written to the log file under certain scenarios.

In the test_invalid_syntax_redhat_feed.py test, even disabling all possible modules before launching the test does not prevent it from falling in the first use case.

The cause of the failure is the large number of messages generated by enabling the internal option wazuh_modules.debug=2, which is required by multiple vulnerability detector tests. When this option is enabled and the manager is restarted, the log file takes up 1MB, this added to the messages generated by the test itself makes FileMonitor unable to process them, causing the error.

**Comment by damarisg** [comment]
Closed by #1605 and #1633.

Test Results after to merge.

| Test Executions | Test |Date  | By  | Status | 
|--|--|--|--|--|
|[YellowMissingFieldRedhat.log](https://github.com/wazuh/wazuh-qa/files/6874391/YellowMissingFieldRedhat.log)| test_feeds/redhat/test_missing_fields_redhat_feed |2021-07-25 | Seyla| :yellow_circle: 
|[YellowMissingFieldRedhat2.log](https://github.com/wazuh/wazuh-qa/files/6874392/YellowMissingFieldRedhat2.log))| test_feeds/redhat/test_missing_fields_redhat_feed |2021-07-25 | Seyla | :yellow_circle: 
|[YellowMissingFieldRedhat3.log](https://github.com/wazuh/wazuh-qa/files/6874393/YellowMissingFieldRedhat3.log)| test_feeds/redhat/test_missing_fields_redhat_feed| 2021-07-25 | Seyla| :yellow_circle: 
|[YellowMissingFieldMSU.log](https://github.com/wazuh/wazuh-qa/files/6874394/YellowMissingFieldMSU.log)| test_feeds/msu/test_missing_fields_msu_feed |2021-07-25 | Seyla| :yellow_circle: 
|[YellowMissingFieldMSU2.log](https://github.com/wazuh/wazuh-qa/files/6874395/YellowMissingFieldMSU2.log)| test_feeds/msu/test_missing_fields_msu_feed |2021-07-25 | Seyla | :yellow_circle: 
|[YellowMissingFieldMSU3.log](https://github.com/wazuh/wazuh-qa/files/6874390/YellowMissingFieldMSU3.log)| test_feeds/msu/test_missing_fields_msu_feed| 2021-07-25 | Seyla| :yellow_circle:
